app/src/pages/attachment/index.tsx

```diff
@@ -1,34 +1,43 @@
 import React from 'react';
 import Link from 'next/link';
+import { S3ApiClient } from '../../utils/clients/S3ApiClient';
+import { IS3Object } from '../../models/data/IS3Object';
 
 const Attachment = () => {
-    const uploadPhoto = async (e: React.ChangeEvent<HTMLInputElement>) => {
-        const file = e.target.files?.[0]!;
-        const filename = encodeURIComponent(file.name);
-        const fileType = encodeURIComponent(file.type);
+    const s3ApiClient = new S3ApiClient();
```

Comment:
The way this is now, it'll instantiate a new S3ApiClient on every rerender (up to 60 times per second)

Should this be `const s3ApiClient = useMemo(() => new S3ApiClient(), [])` so that it is created once per page load?

Even better, you could make a hook (like the existing `useReportClient`) so that it's easier to use :)